fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/jobs/updateloanarrearsageing/LoanArrearsAgeingUpdateHandler.java

```diff
@@ -0,0 +1,202 @@
+    }
+
+    private List<String> updateLoanArrearsAgeingDetailsWithOriginalScheduleForAllLoans() {
+        List<String> insertStatement = new ArrayList<>();
+        String sqlForLoanIdentifiers = buildQueryForLoanIdentifiersWithOriginalSchedule(Boolean.TRUE);
+        List<Long> loanIds = this.jdbcTemplate.queryForList(sqlForLoanIdentifiers, Long.class);
+        if (!loanIds.isEmpty()) {
+            String loanIdsAsString = loanIds.toString();
+            loanIdsAsString = loanIdsAsString.substring(1, loanIdsAsString.length() - 1);
+            Map<Long, List<LoanSchedulePeriodData>> scheduleDate = loanArrearsAgingService.getScheduleDate(loanIdsAsString);
+
+            List<Map<String, Object>> loanSummary = getLoanSummary(loanIdsAsString);
+            loanArrearsAgingService.updateScheduleWithPaidDetail(scheduleDate, loanSummary);
+            loanArrearsAgingService.createInsertStatements(insertStatement, scheduleDate, true);
+        }
+
+        return insertStatement;
+    }
+
+    private String buildQueryForLoanIdentifiersWithOriginalSchedule(Boolean isForAllLoans) {
+        final StringBuilder loanIdentifier = new StringBuilder();
+        loanIdentifier.append("select ml.id as loanId FROM m_loan ml  ");
+        loanIdentifier.append("INNER JOIN m_loan_repayment_schedule mr on mr.loan_id = ml.id ");
+        loanIdentifier.append(
+                "inner join m_product_loan_recalculation_details prd on prd.product_id = ml.product_id and prd.arrears_based_on_original_schedule = true  ");
+        loanIdentifier.append("WHERE ml.loan_status_id = 300 ");
+        if (!isForAllLoans) {
+            loanIdentifier.append(" and ml.id IN (?)");
+        }
+        loanIdentifier.append(" and mr.completed_derived is false  and mr.duedate < ")
+                .append(sqlGenerator.subDate(sqlGenerator.currentBusinessDate(), "COALESCE(ml.grace_on_arrears_ageing, 0)", "day"))
+                .append(" group by ml.id");
+        return loanIdentifier.toString();
+    }
+
+    private List<Map<String, Object>> getLoanSummary(final String loanIdsAsString) {
+        final StringBuilder transactionsSql = new StringBuilder();
+        transactionsSql.append("select ml.id as loanId, ");
+        transactionsSql
+                .append("ml.principal_repaid_derived as principalAmtPaid, ml.principal_writtenoff_derived as  principalAmtWrittenoff, ");
+        transactionsSql.append(" ml.interest_repaid_derived as interestAmtPaid, ml.interest_waived_derived as interestAmtWaived, ");
+        transactionsSql.append("ml.fee_charges_repaid_derived as feeAmtPaid, ml.fee_charges_waived_derived as feeAmtWaived, ");
+        transactionsSql
+                .append("ml.penalty_charges_repaid_derived as penaltyAmtPaid, ml.penalty_charges_waived_derived as penaltyAmtWaived ");
+        transactionsSql.append("from m_loan ml ");
+        transactionsSql.append("where ml.id IN (").append(loanIdsAsString).append(") order by ml.id");
```

Comment:
Nah, lets not append the String directly.

Reply:
Rafactored now.